app/app.go

```diff
@@ -708,9 +746,24 @@ func NewIxoAnteHandler(app *IxoApp, encodingConfig params.EncodingConfig) sdk.An
 	// one, since the default does not allow ED25519 signatures. Thus, like this
 	// we enable ED25519 (as well as Secp) signing of standard Cosmos messages.
 
-	cosmosAnteHandler := authante.NewAnteHandler(
-		app.AccountKeeper, app.BankKeeper, ixotypes.IxoSigVerificationGasConsumer,
-		encodingConfig.TxConfig.SignModeHandler())
+	// TODO: Fix this and figure out what happens here
+	// cosmosAnteHandler, err := authante.NewAnteHandler(
+	// 	app.AccountKeeper, app.BankKeeper, ixotypes.IxoSigVerificationGasConsumer,
+	// 	encodingConfig.TxConfig.SignModeHandler())
+
+	options := authante.HandlerOptions{
+		AccountKeeper:   app.AccountKeeper,
+		BankKeeper:      app.BankKeeper,
+		FeegrantKeeper:  app.FeeGrantKeeper,
+		SignModeHandler: encodingConfig.TxConfig.SignModeHandler(),
+		SigGasConsumer:  ixotypes.IxoSigVerificationGasConsumer,
+	}
+
+	cosmosAnteHandler, err := authante.NewAnteHandler(options)
+
+	if err != nil {
+		panic(sdkerrors.Wrap(err, "could not create Cosmos AnteHandler"))
+	}
```

Comment:
@ig-shaun @demondayza @steveixo 
The  `NewAnteHandler` function now requires more arguments, namely the `FeegrantKeeper: app.FeeGrantKeeper,` Again how should this be handled in the context of IXO?

Reply:
Is the Feegrantkeeper defined anywhere in the cosmos documentation?

Reply:
@demondayza This is the formal docs that I could find regarding the feegrant.
https://docs.cosmos.network/master/modules/feegrant/

I assume you will learn more by going through the code on github.

Reply:
This will require some architectural design decisions, however its a important module.